src/meta-srv/src/bootstrap.rs

```diff
@@ -253,3 +260,11 @@ async fn create_etcd_client(opts: &MetasrvOptions) -> Result<Client> {
         .await
         .context(error::ConnectEtcdSnafu)
 }
+
+async fn create_postgres_client(opts: &MetasrvOptions) -> Result<tokio_postgres::Client> {
+    let postgres_url = opts.store_addrs.first().context(EmptyServerAddrSnafu)?;
```

Comment:
I think it support failover at the making connection phase - it will try all the addresses until make a successful connection. `postgresql://user@host1:1234,host2,host3:5678?`
https://github.com/sfackler/rust-postgres/blob/d2634a46b5bc1de5176f0766626e7a9f6757f362/tokio-postgres/src/connect.rs#L86C1-L133C1

I dont think it support advanced failover. I think we could achieve failover if it connect to a proxy layer such as haproxy or enovy. Or we could implement our connection pool that has the failover feature.